addons/sale/models/sale_order.py

```diff
@@ -1994,3 +1994,6 @@ def _get_lang(self):
             return self.partner_id.lang
 
         return self.env.lang
+
+    def _validate_zero_amount_orders(self):
+        return True
```

Comment:
as commented by VFE, no docstring is needed for the auto-invoice task in website_sale since the code is self-explanatory and similar to the one in website_sale.